### PR TITLE
Expose GeNN preferences to PyGeNN

### PR DESCRIPTION
**Pull request by neworderofjamie**
This lets you access any GeNN preferences from PyGeNN in a future-proof way e.g.

```python
genn_model.GeNNModel("float", "va_benchmark", enableBitmaskOptimisations=True)
```

### CODE REVIEW

File: pygenn/genn_model.py [hunk 1]
```diff
@@ -135,17 +136,10 @@ def __init__(self, precision=None, model_name="GeNNModel",
...
-
-    @selected_gpu.setter
-    def selected_gpu(self, v):
-        if self.use_backend == "CUDA":
-            self._selected_gpu = v
-        elif v is not None:
-            raise Exception("Selecting a GPU is only compatible with the CUDA backend "
-                            "but the {} backend was chosen.".format(self.use_backend))
+        # For backward compatibility, if selected GPU is set, add preferences
+        if selected_gpu is not None:
```
[the remark this answers is not in the document]
Comment (neworderofjamie) [reply pygenn/genn_model.py]: excellent point!

**Comment by neworderofjamie** [comment]
What do you think @chanokin ?

**Comment by chanokin** [comment]
seems good except for a `print("MERRR")` in line 141 :laughing:

**Comment by neworderofjamie** [comment]
Hahah and **this** is why code review is good 😄

**Pull request merged**